packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -46,26 +43,80 @@ expect val INVALID_PROPERTY_KEY: PropertyKey
 @JvmInline
 value class RegistrationToken(val value: Long)
 
+// Pure marker interfaces corresponding to the C-API realm_x_t struct types
+interface CapiT
+interface RealmConfigT : CapiT
+interface RealmSchemaT : CapiT
+interface RealmT : CapiT
+interface LiveRealmT : RealmT
+interface FrozenRealmT : RealmT
+interface RealmObjectT : CapiT
+interface RealmListT : CapiT
+interface RealmResultsT : CapiT
+interface RealmQueryT : CapiT
+interface RealmNotificationTokenT : CapiT
+interface RealmChangesT : CapiT
+interface RealmObjectChangesT : RealmChangesT
+interface RealmCollectionChangesT : RealmChangesT
+// Public type aliases binding to internal verbose type safe type definitions. This should allow us
+// to easily change implementation details later on.
+typealias RealmNativePointer = NativePointer<out CapiT>
+typealias RealmConfigurationPointer = NativePointer<RealmConfigT>
+typealias RealmSchemaPointer = NativePointer<RealmSchemaT>
+typealias RealmPointer = NativePointer<out RealmT>
+typealias LiveRealmPointer = NativePointer<LiveRealmT>
+typealias FrozenRealmPointer = NativePointer<FrozenRealmT>
+typealias RealmObjectPointer = NativePointer<RealmObjectT>
+typealias RealmListPointer = NativePointer<RealmListT>
+typealias RealmResultsPointer = NativePointer<RealmResultsT>
+typealias RealmQueryPointer = NativePointer<RealmQueryT>
+// FIXME Should we differentiate this by generics
+typealias RealmNotificationTokenPointer = NativePointer<RealmNotificationTokenT>
+typealias RealmChangesPointer = NativePointer<RealmChangesT>
+typealias RealmObjectChangesPointer = NativePointer<RealmObjectChangesT>
+typealias RealmCollectionChangesPointer = NativePointer<RealmCollectionChangesT>
+
+// Sync types
+// Pure marker interfaces corresponding to the C-API realm_x_t struct types
+interface RealmAppT : CapiT
+interface RealmAppConfigT : CapiT
+interface RealmSyncConfigT : CapiT
```

Comment:
Is it possible to keep `RealmConfigT and `RealmSyncConfigT` completely separate? I would assume it would cause problems when opening the Realm?

Reply:
Turns out I misunderstood what this actually was. Just disregard the above comment.

Reply:
Yeah. These pointers doesn't correspond 1-1 to the user facing `RealmConfiguration` and `SyncConfiguration`. 

A `RealmConfiguration` has a `RealmConfigurationPointer` and all sync related configurations are done through the `RealmConfigurationPointer` by updating its `RealmSyncConfigurationPointer` entry.